tox.ini

```diff
@@ -47,6 +46,8 @@ commands =
 
 [testenv:py3-server]
 description = Runs all Python3-based server unit and functional tests
+setenv =
+    PYTHONPATH = {env:PYTHONPATH}:{toxinidir}/server/lib:{toxinidir}/lib
```

Comment:
I'm not sure (you would probably know better), but I think we need `PYTHONPATH` entries in _both_ places -- we need to add `{toxinidir}/lib` to the path in _all_ environments, and we need to add `{toxinidir}/server/lib` for server testing.

Reply:
I ran all the tests and they are working fine without adding `PYTHONPATH` for them. Earlier as well `PYTHONPATH` was not added in the tox environment. Is it a new requirement? `PYTHONPATH` is also not present for any tests in the `main` branch

Reply:
It's hard to argue against an approach which seems to be working....  😀

The thing is, at the moment, we don't really have a coherent approach to ensuring that "the user" has a properly-set `PATH` and `PYTHONPATH`.  We have several places where we set these environment variables, depending on whether we're in a Server or an Agent context, and whether we're in an interactive, automated, or test environment.

So, if you're happy with the code as proposed (and, if is works as you claim), then I'll say that it's certainly not making anything worse, so we can go with it.

If it all works without tweaking `PYTHONPATH` at all, that's probably better, as it reduces the number of places we have to consider when something goes awry.  Or, if we can get by without changing `tox.ini` at all, that's probably better, too, as I'm not sure about all the various ways in which that file is used, so I'm not in a position to say whether your changes are likely to work in _all_ of them.

The reason you don't see it being set here in the `main` branch is that, in at least some of the `tox` "environments", the test execution has been delegated to a script, and that is where the `PYTHONPATH` is being set.

Reply:
We will need `PYTHONPATH` here, I cannot see a way for now if we can avoid it. I am planning to add it to Jenkins, if anything will fail, then I will tweak the changes, for now, it seems to be working fine :)

> The reason you don't see it being set here in the main branch is that, in at least some of the tox "environments", the test execution has been delegated to a script, and that is where the PYTHONPATH is being set.

yeah right